test/bench/spec/bsonBench.ts

```diff
@@ -1,77 +1,105 @@
-import { Suite } from 'dbx-js-tools/packages/bson-bench';
+import { PerfSendResult, Suite } from 'dbx-js-tools/packages/bson-bench';
 import { join, resolve } from 'path';
-import { writeFile } from 'fs/promises';
-import { LIBRARY_SPEC } from '../granular/common';
+import { writeFile, readFile } from 'fs/promises';
+import { readEnvVars, ALERT_TAG } from '../granular/common';
 
+type Metadata = {
+  improvement_direction: 'up' | 'down';
+};
 const suite = new Suite('bson micro benchmarks');
 const DOCUMENT_ROOT = resolve(`${__dirname}/../../documents`);
+const { library } = readEnvVars();
+const warmup = 1000;
+const iterations = 10_000;
 // Add flat bson encoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'flat_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'serialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
 // Add flat bson decoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'flat_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'deserialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
 // Add deep bson encoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'deep_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'serialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
 // Add deep bson decoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'deep_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'deserialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
 // Add full bson encoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'full_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'serialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
 // Add full bson decoding
 suite.task({
   documentPath: join(DOCUMENT_ROOT, 'full_bson.json'),
-  library: LIBRARY_SPEC,
-  warmup: 1000,
-  iterations: 10_000,
+  library,
+  warmup,
+  iterations,
   operation: 'deserialize',
-  options: {}
+  options: {},
+  tags: [ALERT_TAG]
 });
 
-suite.run().then(
-  () => {
-    const results = suite.results.map(result => {
+suite
+  .run()
+  .then(() => {
+    return readFile(join(__dirname, '..', '..', 'etc', 'cpuBaseline.json'), 'utf8');
```

Comment:
Can you move this into the block below, make it an `async () =>` function and `await` this? then stack traces are a bit nicer, nothing wrong with an async function chained to a promise.